app/Http/Controllers/EmployeeController.php

```diff
@@ -2,82 +2,154 @@
-            'recordsTotal' => intval($totalEmployees),
-            'recordsFiltered' => $filteredEmployees,
-            'data' => $employees
-        ];
-
-        return Response::json($response);
+          $response = [
+              'draw' => intval($draw),
+              'recordsTotal' => intval($totalEmployees),
+              'recordsFiltered' => $filteredEmployees,
+              'data' => $employees
+          ];
+          return Response::json($response);
     }
 
+
     public function store(Request $request){
 
-        $validated = $request->validate([
-            'name' => 'required|max:50|',
-            'position' => 'required|max:50|',
-            'dob' => 'required|date',
-            'email' => 'required|unique:employees|email',
-            'phone' => 'required|max:13',
-             'address' => 'required|max:255'
-            ]
-        );
-
-        Employee::create($validated);
+        $errors = [];
+
+        try {
+            $employeeValidated = $request->validate([
+                'name' => 'required|max:50|',
+                'position' => 'required|max:50|',
+                'dob' => 'required|date',
+                'email' => 'required|unique:employees|email',
+                'phone' => 'required|max:13',
+                 'address' => 'required|max:255',
+                 'company_id' => 'required',
+                ]
+            );
+
+        } catch (Exception $e) {
+            $errors[] = $e->getMessage();
+        }
+
+        try {
+            $bankAccValidated = $request->validate([
+                'beneficiary_name' => 'required',
+                'bank_name' => 'required',
+                'bank_branch' => 'required',
+                'account_no' => 'required|max:9',
+                ]
+            );
+
+        } catch (Exception $e) {
+            $errors[] = $e->getMessage();
+        }
+
+        if ($errors){
+            return Response::json($errors , 422);
+        }
+
+        $employee = Employee::create($employeeValidated);
+        $bankAccValidated['employee_id'] = $employee->id;
+        BankAccount::create($bankAccValidated);
         return response(200);
+
+    }
+
+    public function edit(string $id){
+        $employee = Employee::with('bankAccount' , 'company')->where('id',$id)->get();
+        return Response::json($employee);
     }
 
+
     public function update(Request $request , string $id){
 
-        $validated = $request->validate([
-            'name' => 'required|max:50',
-            'position' => 'required|max:50',
-            'dob' => 'required|date',
-            'email' => [
-                'required',
-                'email',
-                Rule::unique('employees', 'email')->ignore($id),
-            ],
-            'phone' => 'required|max:13',
-            'address' => 'required|max:255'
-        ]);
-        Employee::find($id)->update($validated);
-        return response(null, 200);
+        $errors = [];
+        try {
```

Comment:
use one block of try catch instead of multiple

Reply:
okay sure

Reply:
https://github.com/axnish27/employee-profile-manager/pull/32#discussion_r1771995053 @Raza9798